### PR TITLE
Release/1.0.0

### PR DESCRIPTION
**Pull request by livingrockrises**
preparing for release 0.0.5

list of PRs merged into branch release/0.0.5

### CODE REVIEW

File: packages/web3-auth/src/SocialLogin.tsx [hunk 1]
```diff
@@ -54,25 +54,31 @@ class SocialLogin {
...
-    network: 'mainnet' | 'testnet' = 'testnet'
-  ) {
+  async init(socialLoginDTO?: Partial<SocialLoginDTO>) {
+    const finalDTO: SocialLoginDTO = {
+      chainId: '0x1',
+      whitelistUrls: {},
+      network: 'testnet'
+    }
+    if (socialLoginDTO) {
+      if (socialLoginDTO.chainId) finalDTO.chainId = socialLoginDTO.chainId
```
Comment (tomarsachin2271) [inline packages/web3-auth/src/SocialLogin.tsx:64]: Under what circumstances this chainId needs to be overridden?? and how does 'testnet' value of network field changes the EOA account generated by social login sdk?? 
@AmanRaj1608 @livingrockrises

File: packages/web3-auth/src/types/Web3AuthConfig.ts [hunk 2]
```diff
@@ -1,4 +1,9 @@
 export type DefaultSocialLoginConfig = {
-  defaultChainId: string
   backendUrl: string
 }
+
+export type SocialLoginDTO = {
+  chainId: string
+  whitelistUrls: { [P in string]: string }
```
Comment (tomarsachin2271) [inline packages/web3-auth/src/types/Web3AuthConfig.ts:7]: What does this type means? { [P in string]: string }

**Comment by tomarsachin2271** [comment]
Also in this release lets move the console logs to a utility method and use a debug flag to log the messages in console.

File: packages/web3-auth/src/types/Web3AuthConfig.ts (hunk 2 above)
Comment (AmanRaj1608) [reply packages/web3-auth/src/types/Web3AuthConfig.ts:7]: The whitelistUrls expects an array of domain to signature which is passed to web3Auth on initialisation. So basically array of key string to value string.
Example 
```json
{
	"http://localhost:3000": "signature.....",
	"http://newdomain.com": "signature.....",
}
```

File: packages/web3-auth/src/SocialLogin.tsx (hunk 1 above)
Comment (AmanRaj1608) [reply packages/web3-auth/src/SocialLogin.tsx:64]: - On web3Auth Initialisation they expect a chainId cause they generate a provider and signer related to that chainId only. 
- after sign in with soicals, the web3auth servers to which it redirects changes on what we pass, which changes the EOA.

Comment (tomarsachin2271) [reply packages/web3-auth/src/SocialLogin.tsx:70]: Would this change in chainId change the generated EOA address as well?

Comment (AmanRaj1608) [reply packages/web3-auth/src/SocialLogin.tsx:70]: no chainId is used to give back the provider only

Comment (AmanRaj1608) [reply packages/web3-auth/src/SocialLogin.tsx:70]: network values can be


image

File: packages/account-abstraction/src/SmartAccountAPI.ts [hunk 3]
```diff
@@ -114,6 +114,10 @@ export class SmartAccountAPI extends BaseWalletAPI {
    * @param info
    */
   async createUnsignedUserOp(info: TransactionDetailsForUserOp): Promise<UserOperation> {
+    // not ideal as it will unnecessary increase callGasLimit, unless appliedGasLimit is sent reasonable
+    /*if ((await this.checkWalletDeployed())) {
```
Comment (tomarsachin2271) [inline packages/account-abstraction/src/SmartAccountAPI.ts:118]: Remove these comments if not needed.

File: packages/common/src/Logger.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+
+  /**
+   * Single method to be used for logging purpose.
+   *
+   * @param {any} message Message to be logged
+   */
+   export function _logMessage(message: any) {
+    // Method would be called ifLogsEnabled is true
```
Comment (tomarsachin2271) [inline packages/common/src/Logger.ts:8]: Logger will be a class with static method log with a static state `isLogsEnabled`, which is set by the main package constructor based on users input by calling the static method `enableLogs` on the class.

```
export class Logger {
    static isLogsEnabled: boolean = false;

    static log(message: any) {
      if(this.isLogsEnabled && console.log) {
        console.log(message);
      }
    }

    static enableLogs(_isLogsEnabled: boolean) {
      this.isLogsEnabled = _isLogsEnabled;
    }
}

```
Also i don't see the usage of this method anywhere in the code yet.

File: packages/smart-account/src/SmartAccount.ts [hunk 5]
```diff
@@ -307,24 +315,41 @@ class SmartAccount extends EventEmitter {
...
+    let transactionRequest : TransactionRequest = {...transaction}
 
+    const isDeployed = await this.contractUtils.isDeployed(
+      chainId,
+      this.DEFAULT_VERSION,
+      this.address
+    )
+
+    if(!isDeployed || isDelegate) {
+       customData.appliedGasLimit = ethers.constants.Two.pow(24) // estimateGas for execFromEntryPoint
```
Comment (tomarsachin2271) [inline packages/smart-account/src/SmartAccount.ts:342]: Why hardcoded value here?

Comment (livingrockrises) [reply packages/smart-account/src/SmartAccount.ts:342]: so userOp callGasLimit is estimation on execFromEntryPoint method on wallet.

```
// Review: doesn't work if wallet is not already deployed.
    const callGasLimit =
      parseNumber(detailsForUserOp.gasLimit) ??
      (await this.provider.estimateGas({
        from: this.entryPoint.address,
        to: this.getWalletAddress(),
        data: callData
      }))
```
      
 so if wallet is not deployed it just estimates and returns 21000 + data cost  then it fails
 
 Further changes is marked for testing / brainstorming and to be launched in next release before gasless goes live on 
 mainnet.

File: packages/account-abstraction/src/SmartAccountAPI.ts (hunk 3 above)
Comment (livingrockrises) [reply packages/account-abstraction/src/SmartAccountAPI.ts]: done

File: packages/common/src/Logger.ts (hunk 4 above)
Comment (livingrockrises) [reply packages/common/src/Logger.ts:8]: acknowledged and noted for logger ticket scope as discussed

**Pull request merged**